### PR TITLE
Generate Holder struct for TYPE declaration with simple type

### LINKED ISSUE #167: Generate Holder struct for `TYPE` declaration with simple type
From #130 and #144

### PR DESCRIPTION
**Pull request by ytanimura**
Resolve #167.

### CODE REVIEW

File: espr/src/ir/type_ref.rs [hunk 1]
```diff
@@ -76,7 +78,7 @@ impl TypeRef {
     pub fn is_simple(&self) -> bool {
         match self {
             TypeRef::SimpleType(..) => true,
-            TypeRef::Named { is_simple, .. } => *is_simple,
+            TypeRef::Named { is_enumerate, .. } => *is_enumerate,
```
Comment (termoshtt) [inline espr/src/ir/type_ref.rs:81]: This is not acceptable without renaming the function `is_simple()` to `is_enumerate`.

File: espr/tests/type_decl.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      dore
+    );
+  END_TYPE;
+
+  ENTITY c_entity;
+    x: REAL;
+  END_ENTITY;
+
+  TYPE c = c_entity;
+  END_TYPE;
```
Comment (termoshtt) [inline espr/tests/type_decl.rs:20]: What did you try to test by this components? It seems that this part is not related to this PR.

File: ruststep/tests/type_decl.rs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+      ENTITY c_entity;
+        x: REAL;
+      END_ENTITY;
+
+      TYPE c = c_entity;
+      END_TYPE;
+
+      ENTITY d;
+        b: b;
```
Comment (termoshtt) [inline ruststep/tests/type_decl.rs:30]: We should add `a: a` here to test non-simple component

File: espr/tests/type_decl.rs (hunk 2 above)
Comment (ytanimura) [reply espr/tests/type_decl.rs:20]: This is not a direct change, but since we have changed the specification for type_decl, we need to add a test to see if there are any changes to the existing items. In other words, the use of placeholders for entity renaming should be tested here. If anything, the missing point is that I have not tested whether placeholders are used when renaming a or b. I will add that now.
